### ISSUE TITLE
[0.9.4.1] Drop Measurement for 1 DB drops all measurements of same name

### ISSUE THREAD

**Issue by simcap** [influxdb/influxdb#4303]
Hi,

I have a measurement called `p2p` in my db `viewers`. It happens that I also have a `testviewers` db with also a measurement `p2p`.

I wanted to clean my `testviewers` db and drop the `p2p` measurement. I did the following steps:
```
> use testviewers
Using database testviewers
> drop measurement p2p
> select count(value) from p2p
> 
```
So it is empty. The issue is that it also cleaned the db `viewers`!! Beforehand this db `viewers` had a lot of data:
```
> select count(value) from p2p
name: p2p
---------
time      count
1970-01-01T00:00:00Z  1.042289e+06
```

Now it is empty:
```
> use viewers
> select count(value) from p2p
name: p2p
---------
time			count
1970-01-01T00:00:00Z	

or 

> select * from p2p
> 
```

Is there a `from` clause where we should specify the db from which to drop the measurement. Or it is that it just wipe a measurement across all dbs?

**Comment by simcap** [comment]
Actually the measurement `p2p` from the `viewers` db came back!! (a bit more data since we have live customers pushing data)
```
> select count(value) from p2p
name: p2p
---------
time      count
1970-01-01T00:00:00Z  1.042318e+06
```

I guess it might be a sync or display issue in the CLI.

**Comment by rkuchan** [comment]
@simcap, thank you for pointing this out. That is definitely unexpected behavior. Interestingly, I'm able to repro the first part with version 0.9.4.1:

```
> create database first
> use first
Using database first
> insert pumpkin orange=1
> insert pumpkin orange=2
> insert pumpkin orange=3
> create database second
> use second
Using database second
> insert pumpkin orange=11
> insert pumpkin orange=12
> insert pumpkin orange=13
> select * from first."default".pumpkin
name: pumpkin
-------------
time				orange
2015-10-02T19:18:22.026149413Z	1
2015-10-02T19:18:23.487714562Z	2
2015-10-02T19:18:24.668921664Z	3

> select * from second."default".pumpkin
name: pumpkin
-------------
time				orange
2015-10-02T19:18:42.130289879Z	11
2015-10-02T19:18:43.805793895Z	12
2015-10-02T19:18:45.448094234Z	13

> use second
Using database second
> drop measurement pumpkin
> select * from second."default".pumpkin
> select * from first."default".pumpkin
>
```

but I can't seem to repro the second part:
```
> use first
Using database first
> select * from pumpkin
> insert pumpkin orange=4
> select * from pumpkin
name: pumpkin
-------------
time				orange
2015-10-02T19:28:18.147674321Z	4
```

**Comment by otoolep** [comment]
This is bad.

If a measurement name is shard across two databases, drop them from 1 drops them from the other.

**Comment by otoolep** [comment]
Thanks for the report @simcap -- we definitely need to fix this ASAP.

**Comment by otoolep** [comment]
The bz1, and presumably b1, engines suffer from this. However the tsm1 engine does not!

**Comment by otoolep** [comment]
This may actually be fixed by moving to tsm1 engine when it is released.

**Comment by simcap** [comment]
@rkuchan The second part I added (see my second comment) was to say that when I logged out of the shell and logged back in the measurement came back (so the drop only wipe one db as expected). So I was thinking more of a display sync issue in the shell.

But as @otoolep points out some engines seems to suffer from this issue. 

Anyway thanks for the promptness, and before doing a lot of work let's ensure it is not just a shell issue.
